server/main.go

```diff
@@ -0,0 +1,198 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const fcct = "./fcct-x86_64-unknown-linux-gnu"
+
+// Variables set via flag
+var (
+	// debug is if we should be printing out debug logs
+	debug bool
+	// host is the host to listen on
+	host string
+	// port is the port to listen on
+	port string
+)
+
+type postData struct {
+	ConfigString string `json:"config_string"`
+}
+
+type responseData struct {
+	Success        bool                   `json:"success"`
+	IgnitionConfig map[string]interface{} `json:"ignition_config"`
+	ErrLines       []int                  `json:"err_lines"`
+	ErrMsg         string                 `json:"err_msg"`
+}
+
+// write the response and sets the status
+func writeResponse(w *http.ResponseWriter, res responseData, status int) {
+	resJSON, err := json.Marshal(res)
+	if err != nil {
+		http.Error(*w, "Failed to parse struct `responseData` into JSON object", http.StatusInternalServerError)
+	}
+
+	(*w).Header().Set("Content-Type", "application/json")
+	(*w).WriteHeader(status)
+	(*w).Write(resJSON)
+}
+
+// runCommand runs the command and returns the error, stdout, and stderr
+func runCommand(command string, input string) (string, string, error) {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd := exec.Command("bash", "-c", command)
```

Comment:
Why run with bash? It's possible that the command could be crafted to inject scripting.

Reply:
Sounds not secure, so in this case just `exe.Command("./fcct", "input_config")` would be better?

Reply:
I believe it would be better.

Reply:
Updated